presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcess.java

```diff
@@ -136,7 +137,15 @@ public synchronized void start()
 
         // getServerInfoWithRetry will return a Future on the getting the ServerInfo from the native process, we intentionally block on the Future till
         // the native process successfully response the ServerInfo to ensure the process has been launched and initialized correctly.
-        getServerInfoWithRetry().get();
+        try {
+            getServerInfoWithRetry().get();
+        }
+        catch (Throwable t) {
+            // If the native process launch failed, it usually indicates the current host machine is overloaded, we need to throw a fatal error to let
+            // Spark shutdown current executor and fail over to another one (Here is the definition of scala fatal error Spark is relying on:
+            // https://www.scala-lang.org/api/2.13.3/scala/util/control/NonFatal$.html)
+            throw new InternalError(t.getMessage(), t.getCause());
```

Comment:
Will isFatalException return true for InternalError? Would you explain why?

What will be the error message? Can we make it say clearly that we started the process, but failed to ping it? Should try to kill the process?

Reply:
`InternalError` is a subclass of VirtualMachineError (abstract class) which is fatal.  Actually I changed mind to created a new exception called `PrestoSparkFatalException` (inherit from `VirtualMachineError`, compared to our existing `PrestoSparkException` which is non-fatal) to make it more explicit.